File: src/mffs/container/ContainerFortronCapacitor.java
package mffs.container;

import mffs.base.ContainerBase;
import mffs.slot.SlotBase;
import mffs.slot.SlotCard;
import mffs.tileentity.TileEntityFortronCapacitor;
import net.minecraft.entity.player.EntityPlayer;

public class ContainerFortronCapacitor extends ContainerBase
{
	private TileEntityFortronCapacitor tileEntity;

	public ContainerFortronCapacitor(EntityPlayer player, TileEntityFortronCapacitor tileEntity)
	{
		super(tileEntity);
		this.tileEntity = tileEntity;

		// Frequency Card
		this.addSlotToContainer(new SlotCard(this.tileEntity, 0, 9, 74));
		this.addSlotToContainer(new SlotCard(this.tileEntity, 1, 27, 74));

		// Upgrades
		this.addSlotToContainer(new SlotBase(this.tileEntity, 2, 154, 47));
		this.addSlotToContainer(new SlotBase(this.tileEntity, 3, 154, 67));
		this.addSlotToContainer(new SlotBase(this.tileEntity, 4, 154, 87));

		this.addPlayerInventory(player);
	}
}

File: src/main/scala/mffs/production/ContainerFortronCapacitor.scala
package mffs.production

import mffs.item.card.ItemCardFrequency
import mffs.slot.SlotBase
import net.minecraft.entity.player.EntityPlayer
import resonant.lib.gui.ContainerBase
import resonant.lib.prefab.slot.SlotSpecific

class ContainerFortronCapacitor(player: EntityPlayer, tileEntity: TileFortronCapacitor) extends ContainerBase(tileEntity)
{
  //Frequency
  addSlotToContainer(new SlotSpecific(tileEntity, 0, 8, 114, classOf[ItemCardFrequency]))

  //Upgrade slots
  (0 to 2) foreach (y => addSlotToContainer(new SlotBase(tileEntity, y + 1, 154, 47 + y * 18)))

  //Input slots
  for (x <- 0 to 1; y <- 0 to 1)
    addSlotToContainer(new SlotBase(this.tileEntity, x + y * 2 + 4, 9 + x * 18, 74 + y * 18))

  //Output slots
  for (x <- 0 to 1; y <- 0 to 1)
    addSlotToContainer(new SlotBase(this.tileEntity, x + y * 2 + 8, 91 + x * 18, 74 + y * 18))

  addPlayerInventory(player)
}